compiler-cli/src/beam_compiler.rs

```diff
@@ -64,19 +64,35 @@ impl BeamCompiler {
         })?;
 
         let mut buf = String::new();
+        let mut accumulated_modules: Vec<String> = Vec::new();
         while let (Ok(_), Ok(None)) = (inner.stdout.read_line(&mut buf), inner.process.try_wait()) {
-            match buf.trim() {
-                "gleam-compile-result-ok" => return Ok(()),
-                "gleam-compile-result-error" => {
-                    return Err(Error::ShellCommand {
-                        program: "escript".into(),
-                        err: None,
-                    })
+            let trimmed = buf.trim();
+
+            if trimmed.starts_with("gleam-compile") {
```

Comment:
Would be nice to flatten this into just a `match` with `if` guards to avoid the extra indentation